[WIP] feat: Add windows pipelines (#332)
kubevirt/kubevirt-tekton-tasks: templates-pipelines/windows-customize/README.md

# Windows Customize Pipeline

> This pipeline clones the DataVolume of a basic and generalized Windows 10, 11 or Server 2k22 installation and runs arbitrary customization commands through an unattend.xml after startup of the VirtualMachine. As an example a ConfigMap which installs Microsoft SQL Server Express and generalizes the VirtualMachine after is included (`windows-sqlserver`) or a ConfigMap that install VSCode (`windows-vs-code`).

Reviewer: ```suggestion
This pipeline clones the DataVolume of a basic and generalized Windows 10, 11 or Server 2k22 installation and runs arbitrary customization commands through an unattend.xml after startup of the VirtualMachine. As an example a ConfigMap which installs Microsoft SQL Server Express and generalizes the VirtualMachine after (`windows-sqlserver`) and a ConfigMap that installs VSCode (`windows-vs-code`) are included.
```

For basic setup after the first start of a customized VirtualMachine an example unattend.xml is included in the pipeline's ConfigMap `windows10-unattend`, or `windows11-unattend`.

This example pipeline can be used for running Windows 10, 11 or Server 2k22 (or others - not tested!). Always adjust pipeline parameters for the Windows version you are currently using (e.g. differe name, different base image name, etc.). It is possible to use `windows-sqlserver` ConfigMap for Windows 11 or Server 2k22 and vice versa (`windows-vs-code` for Windows 10 or Server 2k22).

The provided reference ConfigMap (`windows-sqlserver`) boots Windows 10, 11 or Windows Server 2k22 into Audit mode, applies the customizations as part of Powershell script (ran by `SynchronousCommand`) and then generalizes the VirtualMachine again. The Powershell script can be adapted as desired to apply other customizations.

## Prerequisites

- KubeVirt `>=v1.0.0`

> - Tekton Pipelines `>=v0.44.0`

Reviewer: Here it says v0.44.0, in the task yaml it says 0.43.0?

3. `wait-for-vmi-status` task waits until the VirtualMachine shuts down.
4. `cleanup-vm` deletes the installer VirtualMachine (also in case of failure of the previous tasks).
5. The output artifact will be the `win*-customized` DataVolume with the customized Windows installation. It will boot into the Windows OOBE and needs to be setup further before it can be used (depends on the applied customizations).
6. The `windows11-unattend` ConfigMap can be used to boot the VirtualMachine into the Desktop (depends on the applied customizations).

## How to run

Pipeline runs with resolvers:
{% for item in pipeline_runs_yaml %}
```yaml
oc create -f - <<EOF
{{ item | to_nice_yaml }}EOF
```
{% endfor %}

### Usage in multiple namespaces

When a user defines a different namespace in e.g. `baseDvNamespace`, then the serviceAccount under which the pipeline is running will require additional permissions in that namespace. Required permissions to run task in different namespace can be found in README of each task.

> ## Cancelling/Deleteting pipelineRuns

Reviewer: ```suggestion
## Cancelling/Deleting PipelineRuns
```
